modules/tinymce/src/themes/silver/main/ts/ui/toolbar/CommonToolbar.ts

```diff
@@ -39,20 +44,34 @@ export interface MoreDrawerToolbarSpec extends ToolbarSpec {
 
 export interface ToolbarGroup {
   readonly title: Optional<string>;
+  readonly label: Optional<string>;
   readonly items: AlloySpec[];
 }
 
 const renderToolbarGroupCommon = (toolbarGroup: ToolbarGroup) => {
-  const attributes = toolbarGroup.title.fold(() => ({}),
-    (title) => ({ attributes: { title }}));
+  const attributes = toolbarGroup.label.isNone() ? toolbarGroup.title.fold(() => ({}),
+    (title) => ({ attributes: { title }})) : toolbarGroup.label.fold(() => ({}),
+    (label) => ({ attributes: { name: label }}));
+
   return {
     dom: {
-      tag: 'div',
-      classes: [ 'tox-toolbar__group' ],
+      tag: toolbarGroup.label.isSome() ? 'fieldset' : 'div',
```

Comment:
I'm not sure that fieldset is the right thing to use since from what I understand it's more about getting various input fields in form together. Since this already has a `role=toolbar` I think just a div would work for screen readers but with the additional aria attributes. We can just stick the `aria-label` on the div then have `role=presentation` on the visual label making that a label element. Or make the visual label something that the screen readers pick up by a element that the toolbar references by a `aria-labeled-by`.

Reply:
I can see how this happened, it's using a `legend` for the label. But it's designed for input elements, not buttons.

- [ ] revert both `fieldset` and `legend` to `div`

Reply:
reverted